.githooks/pre-commit

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -euxo pipefail
+
+KT_FILES="$(git --no-pager diff --name-status --no-color --cached | awk '$1 != "D" && $2 ~ /\.kts|\.kt/ { print $2}')"
+
+if [ -z "$KT_FILES" ];
+then
+  echo "No Kotlin staged files."
+else
+  echo "Running ktlint over these files:"
+  echo "$KT_FILES"
+
+  ./gradlew ktlintFormat -PinternalKtlintGitFilter="$KT_FILES"
+
+  echo "Completed ktlint run."
+
+  echo "Starting detekt run"
+  ./gradlew detekt
+  echo "Completed detekt run"
+
+  echo "$KT_FILES" | while read -r file; do
+      if [ -f $file ]; then
+          git add $file
```

Comment:
Files get commited if this is removed, too. So why is this needed?

Reply:
re-adding the file is necessary to make the changes done by kt-lint part of the ongoing commit. Otherwise just  the previously staged status of the code (pre-kt-lint) is committed and an additional manually executed commit would be necessary to actually commit the kt-lint changes.